spec/controllers/facilitator/projects_controller_spec.rb

```diff
@@ -3,64 +3,161 @@
+          post :create, project_creation_form: params
+          is_expected.to render_template(:new)
+        end
+      end
+    end
+
+    describe '#edit' do
+      context 'facilitator access edit page' do
+        let(:project) { create(:project) }
+
+        before { get :edit, { id: project.id } }
+
+        it 'sets the project modification form with the selected project' do
+          expect(assigns(:project_modification_form).project).to eq(project)
+        end
 
+        it 'renders the edit page' do
+          is_expected.to render_template(:edit)
+        end
       end
+    end
 
+    describe '#update' do
+      let(:project_modification_form) { instance_double(ProjectModificationForm) }
 
+      context 'facilitator updates project' do
+        let(:project) { create(:project) }
+        let(:student_attributes) { { students_attributes: attributes_for(:student) } }
+        let(:project_attributes) { attributes_for(:project).merge(student_attributes) }
+
+        it 'redirects to root path and shows a success flash message' do
+          params = { project_attributes: {
+              name: "Renamed Project 1",
+              location: "Location X",
+              start_on: "2017-06-21",
+              end_on: "2017-10-18",
+              status: "1",
+              students_attributes: {
+                  name: "Student1",
+                  birthday: "2016-11-06",
+                  class_name: "1A",
+                  comments: nil
+              }}}
+
+          allow(ProjectModificationForm).to receive(:new).with(project, params) { project_modification_form }
+          allow(project_modification_form).to receive(:save) { true }
+          post :update, id: project.id,  project_modification_form: params
+
+          is_expected.to redirect_to(root_path)
+          is_expected.to set_flash[:success]
+        end
+
+        it 'renders edit page on ununsuccessful update, shows error message, and has object' do
+          params = { project_attributes: {
+              name: "Renamed Project 1",
+              location: "Location X",
+              start_on: "2017-06-21",
+              end_on: "2017-10-18",
+              status: "1",
+              students_attributes: {
+                  name: "Student1",
+                  birthday: "2016-11-06",
+                  class_name: "1A",
+                  comments: nil
+              }}}
+
+          allow(ProjectModificationForm).to receive(:new).with(project, params) { project_modification_form }
+          allow(project_modification_form).to receive(:save) { false }
+          post :update, id: project.id,  project_modification_form: params
+
+          is_expected.to render_template(:edit)
+          expect(assigns(:project_modification_form)).to eq(project_modification_form)
+        end
+
+      end
     end
 
   end
 
 end
+
+def stringify_hash_values(hash)
+  hash.keys.each {|k| hash[k] = hash[k].to_s}
+end
```

Comment:
Moved it out, I used require_relative as I didn't want to mess with the load path of the file, let me know if you know any better solution